src/drawing-tool/components/runtime.tsx

```diff
@@ -140,7 +144,66 @@ export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, set
     }
   };
 
+  const handleUploadBtnClick = () => {
+    setUploadControlsVisible(true);
+  };
+  
+  const uploadFile = (fileOrUrl: File | string) => {
+    setUploadControlsVisible(false);
+    setUploadInProgress(true);
+    // Always copy image to S3 using Shutterbug, so it works even when the image disappears from the external location.
+    // Local image could be theoretically stored as dataSrc but it might be too big for Firestore storage that
+    // is used by ActivityPlayer. So, copying to S3 is a safer option.
+    let uploadPromise;
+    if (typeof fileOrUrl === "string") {
+      uploadPromise = copyImageToS3(fileOrUrl);
+    } else {
+      uploadPromise = copyLocalImageToS3(fileOrUrl);
+    }
+    uploadPromise
+      .then(url => {
+        handleSetInteractiveState({ userBackgroundImageUrl: url });
+      })
+      .catch((error) => {
+        window.alert(error);
+        console.error(error);
+      })
+      .finally(() => {
+        setUploadInProgress(false);
+      });
+  };
+
+  const handleFileUpload = () => {
+    const file = fileInputRef.current?.files?.[0];
+    if (file) {
+      uploadFile(file);
+    }
+  };
+
+  const handleFileDrop = (event: React.DragEvent) => {
+    const file = event.dataTransfer?.files?.[0];
+    if (file) {
+      // Local file dropped.
+      uploadFile(file);
+    } else {
+      // URL dragged.
+      const items = event.dataTransfer?.items;
+      const item = Array.from(items).find(i => i.kind === "string" && i.type.match(/^text\/uri-list/));
+      item?.getAsString(src => {
+        uploadFile(src);
+      });
+    }
+    cancelEvent(event);
+  };
+
+  // Necessary to make file drop work.
+  const cancelEvent = (event: React.DragEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+  };
```

Comment:
It's nice and easy, although probably we don't want to use this approach in more places because of the security reasons. Here's some discussion about it:
https://concord-consortium.slack.com/archives/C0M5CM1RA/p1602009683127900?thread_ts=1601919521.110500&cid=C0M5CM1RA
Generally, we shouldn't store user data relying on security through obscurity which is the core of the Shutterbug design. 😉 We just do it here, as LARA / image question is relying on Shutterbug anyway. But @scytacki didn't want to use it in new places I think.

Reply:
Ah, yes -- now that you mention it I think we didn't use a Shutterbug-based approach in CLUE for that very reason.